core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts

```diff
@@ -643,6 +684,246 @@ describe('DotEmaShellComponent', () => {
+                    }
+                });
+
+                spectator.detectChanges();
+
+                tick(1000);
+                spectator.detectChanges();
+
+                const confirmDialog = spectator.query(byTestId('confirm-dialog'));
+
+                confirmDialog
+                    .querySelector('.p-confirm-dialog-reject')
+                    .dispatchEvent(new MouseEvent('click'));
+
+                expect(router.navigate).toHaveBeenCalledWith([], {
+                    queryParams: { language_id: 1 },
+                    queryParamsHandling: 'merge'
+                });
+            }));
+
+            it('should open a dialog to create the page in the new language when the user accepts the creation', fakeAsync(() => {
+                spectator.triggerNavigation({
+                    url: [],
+                    queryParams: {
+                        language_id: 2,
+                        url: 'index',
+                        'com.dotmarketing.persona.id': DEFAULT_PERSONA.identifier
+                    }
+                });
+
+                spectator.detectChanges();
+
+                const dialog = spectator.component.dialog;
+
+                const translatePageSpy = jest.spyOn(dialog, 'translatePage');
+
+                tick(1000);
+                spectator.detectChanges();
+
+                const confirmDialog = spectator.query(byTestId('confirm-dialog'));
+
+                confirmDialog
+                    .querySelector('.p-confirm-dialog-accept')
+                    .dispatchEvent(new MouseEvent('click'));
+
+                expect(translatePageSpy).toHaveBeenCalledWith({
+                    newLanguage: 2,
+                    page: {
+                        canEdit: true,
+                        canRead: true,
+                        identifier: '123',
+                        inode: '123',
+                        live: true,
+                        liveInode: '1234',
+                        pageURI: 'index',
+                        stInode: '12345',
+                        title: 'hello world'
+                    }
+                });
+            }));
+
+            it('should open a dialog to create the page and navigate to default language if the user closes the dialog', fakeAsync(() => {
+                spectator.triggerNavigation({
+                    url: [],
+                    queryParams: {
+                        language_id: 2,
+                        url: 'index',
+                        'com.dotmarketing.persona.id': DEFAULT_PERSONA.identifier
+                    }
+                });
+
+                spectator.detectChanges();
+
+                tick(1000);
+                spectator.detectChanges();
+
+                const confirmDialog = spectator.query(byTestId('confirm-dialog'));
+
+                const dialog = spectator.debugElement.query(By.css('[data-testId="ema-dialog"]'));
```

Comment:
@zJaaal I suggest to use directly spectator way `spectator.query(byTestId('ema-dialog'));`.